bip32/pk.go

```diff
@@ -0,0 +1,132 @@
+package bip32
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/libsv/go-bk/base58"
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bk/chaincfg"
+	"github.com/libsv/go-bk/crypto"
+)
+
+var (
+	reNumericPlusTick = regexp.MustCompile(`^[0-9]+'{0,1}$`)
+)
+
+// PublicKey defines a single public key and public address.
+type PublicKey struct {
+	PublicKey []byte
+	Address   string
+}
+
+// ToAddress will return the publicKey bytes for the current extendedkey as well as the address.
+// An appropriate mainnet or testnet address is determined by the ExtendedKey settings.
+//
+// If you are using a Zeroed ExtendedKey, use the ToAddressWithNet method.
+func (k *ExtendedKey) ToAddress() (*PublicKey, error) {
+	return k.ToAddressWithNet(k.IsForNet(&chaincfg.MainNet))
+}
+
+// ToAddressWithNet will return the publicKey bytes for the current extendedkey as well as the address.
+// This takes a mainnet param that if true will return a mainnet address (prefixed with 1), otherwise
+// a testnet address is returned.
+//
+// This method should only be used if you are dealing with a Zeroed ExtendedKey as it
+// cannot determine the network to use. Instead for most instances use ToAddress.
+func (k *ExtendedKey) ToAddressWithNet(mainnet bool) (*PublicKey, error) {
```

Comment:
did we get his function from?

Reply:
Must have been part of the lib already, I've just ported to a different file I think...

Reply:
It used to be `key.Address(&chaincfg.MainNet)`